### PR TITLE
Furniture/Terrain Expansion 1: Mainline and Improve More Locations furniture

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Content "Mainline furniture from More Locations mod, expand with more home furnishings, and implement in some demo maps"```

#### Purpose of change

Partially addresses https://github.com/CleverRaven/Cataclysm-DDA/issues/28187 - adds some mundane things like furnaces, hot water heaters, file cabinets, etc.

#### Describe the solution
Mainlines the following furniture:
- water heater
- air conditioner
- furnace
- piano
- wardrobe
- filing cabinet

Converts the following terrain to furniture (and majorly updates it):
- Industrial chemical mixer
- Robotic assembler
- power lathe
- hydraulic press
- drill press
- arc furnace

Added the following new furniture that I made myself
- glass front cabinets
- standing floor lamps
- utility shelves
- tablesaw
- bandsaw
- mitre saw
- router table
- planer
- jointer

As a side effect of making a realistic deconstruction to filing cabinets, this adds a new item "lock" that can sometimes be salvaged and used to build cargo locks. The assumption is that most filing cabinets have a key either inside or nearby.

Adds most of these furnitures to some of the home basements, the piano to one model of church, and some of the industrial stuff to Public Works and garages. The water heaters spawn with a random amount of water in them... clean drinking water. In basements where it doesn't freeze. I'm going to need to nerf something before I lose street cred.

52834303-13fbaf80-3096-11e9-9082-50358ebd47b6

Addendum: After adding woodworking furniture, I also went through and did a pretty thorough update to the lumberyard, making it contain (and this is shocking I know) an actual large amount of lumber in addition to woodworking tools.

Many more maps could benefit from having someone go through and update them. The following furniture still needs implementation:
- robotic assembler
- chemical mixer
- arc furnace

#### Describe alternatives you've considered
See the original Issue for a full list.

While not all this stuff was in my issue it all fits with the theme of adding more furniture and terrain now, rather than as needed, so that all appropriate buildings can benefit from it.

#### Additional context
- [x] Mainline furniture and terrain as-is
- [x] Add better deconstruction stuff for terrain (may still have more to add)
- [x] Add better deconstruction stuff for furniture
- [x] Add descriptive texts to these items
- [x] Add these items to some common maps
- [x] Playtest
- [x] fix errors I spotted in Public Works last night
- [x] add looks_like to new furniture

### CODE REVIEW

File: data/json/furniture.json [hunk 1]
```diff
@@ -3381,5 +3381,207 @@
...
+    "symbol": "}",
+    "color": "dark_gray",
+    "move_cost_mod": 2,
+    "required_str": 8,
+    "flags": [ "PLACE_ITEM", "TRANSPARENT", "CONTAINER", "BLOCKSDOOR", "MOUNTABLE" ],
+    "deconstruct": {
+      "items": [
+        { "item": "sheet_metal", "count": [ 2, 6 ] },
+        { "item": "scrap", "count": [ 2, 6 ] },
+        { "item": "clockworks", "count": [ 0, 2 ] }
```
Comment (Night-Pryanik) [inline data/json/furniture.json]: Why are these here?

File: data/json/furniture.json [hunk 2]
```diff
@@ -3381,5 +3381,207 @@
...
     }
+  },
+  {
+    "type": "furniture",
+    "id": "f_file_cabinet",
+    "name": "filing cabinet",
+    "description": "A set of drawers in a sturdy metal cabinet, used to hold files.  It can be locked to protect important information.",
+    "symbol": "}",
+    "color": "dark_gray",
+    "move_cost_mod": 2,
```
Comment (Night-Pryanik) [inline data/json/furniture.json]: It is passable?

File: data/json/furniture.json [hunk 3]
```diff
@@ -3381,5 +3381,207 @@
...
+    }
+  },
+  {
+    "type": "furniture",
+    "id": "f_piano",
+    "name": "piano",
+    "description": "The ol' ebony and ivory.",
+    "symbol": "P",
+    "color": "i_black",
+    "move_cost_mod": 2,
```
Comment (Night-Pryanik) [inline data/json/furniture.json]: It is passable? And if so, absolutely no penalties, like moving over grass?

File: data/json/furniture.json [hunk 4]
```diff
@@ -3381,5 +3381,207 @@
...
+    "required_str": -1,
```
Comment (Night-Pryanik) [inline data/json/furniture.json]: I guess piano should be movable.

File: data/json/furniture.json [hunk 5]
```diff
@@ -3381,5 +3381,207 @@
...
+    "flags": [ "FLAMMABLE", "ORGANIC", "BLOCKSDOOR", "MOUNTABLE" ],
+    "deconstruct": { 
+      "items": [ 
+        { "item": "2x4", "count": 12 },
+        { "item": "nail", "charges": [ 8, 12 ] },
+        { "item": "wire", "count": [ 2, 10 ] },
+        { "item": "cable", "count": [ 2, 10 ] },
```
Comment (Night-Pryanik) [inline data/json/furniture.json]: Copper wiring in plastic jacket in piano?

File: data/json/furniture.json (hunk 1 above)
Comment (I-am-Erk) [reply data/json/furniture.json]: There is no lock item, but the lock on the filing cabinet should be exactly the kind of lock you can salvage to make a cargo lock. This is the best I can do to represent that.

File: data/json/furniture.json (hunk 2 above)
Comment (I-am-Erk) [reply data/json/furniture.json]: Looks like that's from the mod. Same with the comment below, looks like it needs updating.

File: data/json/furniture.json (hunk 5 above)
Comment (I-am-Erk) [reply data/json/furniture.json]: Do we have any smaller wire than Wire? I copied that in because it's 'copper wire'.

File: data/json/terrain.json [hunk 6]
```diff
@@ -8453,5 +8453,225 @@
...
+    "name": "drill press",
+    "description": "A powerful drill mounted on a slide that lets it drop precisely down.",
+    "symbol": "7",
+    "color": "yellow_red",
+    "move_cost": 0,
+    "flags": [ "WALL", "PERMEABLE", "NOITEM", "BASHABLE", "DECONSTRUCT" ],
+    "bash": {
+      "str_min": 20,
+      "str_max": 150,
+      "explosive": 1,
```
Comment (Night-Pryanik) [inline data/json/terrain.json]: Why? Does it have some grenade inside?

File: data/json/terrain.json [hunk 7]
```diff
@@ -8453,5 +8453,225 @@
...
+  },
+  {
+    "type": "terrain",
+    "id": "t_ind_drill",
+    "name": "drill press",
+    "description": "A powerful drill mounted on a slide that lets it drop precisely down.",
+    "symbol": "7",
+    "color": "yellow_red",
+    "move_cost": 0,
+    "flags": [ "WALL", "PERMEABLE", "NOITEM", "BASHABLE", "DECONSTRUCT" ],
```
Comment (Night-Pryanik) [inline data/json/terrain.json]: `DECONSTRUCT` flag? AFAIK there is no such flag, I can't find any mention of it in source files, only in `JSON_FLAGS.md`. In `furniture.json` and `terrain.json` there are few occurrences of this flag. I suppose it is either outdated or non-existing.  Maybe you meant `EASY_DECONSTRUCT`?

**Comment by Night-Pryanik** [comment]
I think you need to add these new types of furniture and terrain to vanilla locations, as for now this is a content without representation ingame.

File: data/json/terrain.json [hunk 8]
```diff
@@ -8453,5 +8453,225 @@
...
+  },
+  {
+    "type": "terrain",
+    "id": "t_ind_furnace",
+    "name": "industrial furnace",
+    "description": "Not the kind of furnace you'd heat your house with, this is a device for heating things to extreme temperatures as part of industrial crafting processes.",
+    "symbol": "0",
+    "color": "white_red",
+    "move_cost": 0,
+    "flags": [ "WALL", "BASHABLE", "PERMEABLE", "CONTAINER", "FIRE_CONTAINER", "SUPPRESS_SMOKE", "PLACE_ITEM" ],
```
[the remark this answers is not in the document]
Comment (I-am-Erk) [reply data/json/terrain.json]: Bear in mind with all these questions that I just copy and pasted these from a mod a few minutes ago. I don't know why they were done this way in the first place. I can fix them though.

**Comment by I-am-Erk** [comment]
I could throw an example or two into existing maps as part of this pr, sure. Many are part of the lab update I'm doing, and the few that aren't I just included for completeness... But it would be pretty easy to toss them into a couple maps.

File: data/json/furniture.json (hunk 1 above)
Comment (Night-Pryanik) [reply data/json/furniture.json]: I don't think this is an appropriate solution. Either add a specific item for a lock or remove the clockworks from the items table.

File: data/json/furniture.json (hunk 5 above)
Comment (Night-Pryanik) [reply data/json/furniture.json]: No, but I think steel string (as `wire`) alone will suffice. Or there is copper strings in pianos IRL?

File: data/json/terrain.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (Night-Pryanik) [reply data/json/terrain.json]: Fix, please.

**Comment by SunshineDistillery** [comment]
I commented in the issues, but I'll say it here too.  Some of these terrains (factory stuff) should be furniture instead.  It's just easier to avoid having bash_terrain that doesn't match the rest of the map, setting indoor/roof flags, and things like supporting walls.  Since furniture and terrain are so similar it shouldn't be hard to convert, just set str_min to -1 so they can't be dragged around.  

I made these a long time ago and they're a little crummy.

**Comment by I-am-Erk** [comment]
The only reason not to make them furniture is that I'm mainlining them from the mod, and that will mean the mod has to have duplicate but different copies or be updated. Also, it means the graphical tiles won't apply... But the tiles for the terrain turned out to be pretty inappropriate when I checked last night, so that's okay.

File: data/json/furniture.json (hunk 2 above)
Comment (I-am-Erk) [reply data/json/furniture.json]: It's just as passable as the countertop, which seems about right. The problem here is that the graphic shows something as tall as a locker, while the flags indicate it's probably one of the waist-high two drawer kind.

File: data/json/furniture.json (hunk 3 above)
Comment (I-am-Erk) [reply data/json/furniture.json]: Move cost mod 2 is the same as other flat-topped things like tables and countertops. I think the assumption was it was a close-top grand piano that you could vault over.  That said, I think it should be higher... it sits higher off the ground than those things and may be open and hard to move overtop.

File: data/json/furniture.json (hunk 1 above)
Comment (I-am-Erk) [reply data/json/furniture.json]: I hadn't aimed to add a bunch of new lock stuff, but sure, what's one more tangent... ;)

**Comment by Night-Pryanik** [comment]
Oh, it's really cool you decided to add a heater and a furnace to the basement, looking good!

Other matters: consider adding WIP tag if you're still working on the PR.

**Comment by I-am-Erk** [comment]
OK. This should have a playtest to make sure I didn't mess up any spawn coordinates or anything, but short of that I have no plans to further develop it.

**Comment by Night-Pryanik** [comment]
> I'm going to need to nerf something before I lose street cred.

Just upper cap the amount of water in heaters from 100 to something like 50 or even lower.

**Comment by I-am-Erk** [comment]
I think it should be realistic. Most houses would have 60L of potable water in the basement, limiting it to 25L is already a bit harsh.

It's fairly well balanced by where it's found; houses are in risky areas for new characters. I don't mind having a realistic benefit to going into them. Later on in development, basements will come with their own set of problems.

**Comment by kevingranade** [comment]
Actually, how does one remove water from a heater? Is there a drain valve?

**Comment by I-am-Erk** [comment]
Yep. In fact that very valve is included in the deconstruction recipe!

File: data/json/furniture.json (hunk 2 above)
Comment (Night-Pryanik) [reply data/json/furniture.json]: Then if it's passable, it shouldn't be passable as easy as through the grass.

Comment (I-am-Erk) [reply data/json/furniture.json]: It's the same move cost mod as every other waist-high flat-topped object, including ovens, countertops, tables, sinks, bathtubs, and more. If there's a problem here it's more likely to be the grass.

Comment (Night-Pryanik) [reply data/json/furniture.json]: Okay, I got it. Seems like we're gonna need the global audit of move costs for furniture and terrain after the 0.D.

Comment (I-am-Erk) [reply data/json/furniture.json]: I think what it is is that the terrain move cost and the furniture are added together, so furniture with a move mod of 2 is still higher than terrain with that, but I'm not sure.

File: data/json/furniture.json [hunk 9]
```diff
@@ -3381,5 +3382,750 @@
...
+    "bash": {
+      "str_min": 40,
+      "str_max": 150,
+      "sound_fail": "clang!",
+      "items": [
+        { "item": "cable", "charges": [ 0, 4 ] },
+        { "item": "scrap", "count": [ 8, 12 ] },
+        { "item": "steel_chunk", "count": [ 2, 4 ] },
+        { "item": "steel_lump", "count": [ 1, 2 ] },
+        { "item": "circsaw_blade", "count": [ 4, 6 ] }
```
Comment (KorGgenT) [inline data/json/furniture.json]: are you supposed to get more circular saws by bashing this rather than deconstruction?

File: data/json/furniture.json [hunk 10]
```diff
@@ -3381,5 +3382,750 @@
...
+    "symbol": "{",
+    "looks_like": "f_rack",
+    "description": "A simple heavy-duty plastic and metal shelving unit.",
+    "color": "light_gray",
+    "move_cost_mod": -1,
+    "required_str": 6,
+    "flags": [ "PLACE_ITEM", "BLOCKSDOOR", "EASY_DECONSTRUCT" ],
+    "deconstruct": {
+      "items": [
+        { "item": "sheet_metal_small", "count": [ 0, 4 ] },
```
Comment (KorGgenT) [inline data/json/furniture.json]: i would raise at least the minimum a tad, because bashing it will give you more (on average)

File: data/json/furniture.json (hunk 9 above)
Comment (I-am-Erk) [reply data/json/furniture.json]: You're not supposed to get circular saw blades from a lathe at all. Not sure where those came from.

**Comment by I-am-Erk** [comment]
I think this is ready to go
